src/AutoPilotPlugins/PX4/AirframeComponentController.cc

```diff
@@ -135,8 +135,10 @@ void AirframeComponentController::_rebootAfterStackUnwind(void)
 {    
     _uas->executeCommand(MAV_CMD_PREFLIGHT_REBOOT_SHUTDOWN, 1, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0);
     qgcApp()->processEvents(QEventLoop::ExcludeUserInputEvents);
-    QGC::SLEEP::sleep(1);
-    qgcApp()->processEvents(QEventLoop::ExcludeUserInputEvents);
+    for (unsigned i = 0; i < 2000; i++) {
+        QGC::SLEEP::usleep(500);
+        qgcApp()->processEvents(QEventLoop::ExcludeUserInputEvents);
```

Comment:
Isn't this a huge amount of wait time? 2000 * (500 usecs)? Maybe I'm reading something wrong.